test/integration/projects_test.rb

```diff
@@ -0,0 +1,64 @@
+require 'test_helper'
+
+class ProjectsTest < ActionDispatch::IntegrationTest
+  test "create" do
+    before = Project.all.size
+    Dir.mktmpdir do |tmpdir|
+      params = {
+        project: {name: 
+                  "test project one", 
+                description: 
+                  "test project one description", 
+                directory: 
+                  tmpdir
+                }
+              }
+      post projects_path, params: params
+      follow_redirect!
+      assert_response :success
+      get projects_path
+      assert_response :success
+      after = Project.all.size
+      assert after = before + 1
+      end
+  end
+  test "deleting_a_project" do
```

Comment:
Spaces are an underrated feature in readability. Which is to say - you need spaces to make code readable.

Also you're indentation is off - see my note here at that second end.
```
        post projects_path, params: params
        follow_redirect!
        assert_response :success

        get projects_path
        assert_response :success

        projects_after = Project.all.size
        assert projects_after = projects_before + 1
      end  # watch the ends here. make sure the thing above an end is always indented.
  end

  test "deleting_a_project" do 
```